diff --git a/libavcodec/cbs.c b/libavcodec/cbs.c
index 726cdfb9d7..431675643d 100644
--- a/libavcodec/cbs.c
+++ b/libavcodec/cbs.c
@@ -60,9 +60,15 @@ static const CodedBitstreamType *const cbs_type_table[] = {
 #if CBS_VP9
     &CBS_FUNC(type_vp9),
 #endif
+#if !CBS_APV && !CBS_AV1 && !CBS_H264 && !CBS_H265 && !CBS_H266 && !CBS_LCEVC && !CBS_JPEG && !CBS_MPEG2 && !CBS_VP8 && !CBS_VP9
+    NULL,
+#endif
 };
 
 const enum AVCodecID CBS_FUNC(all_codec_ids)[] = {
@@ -105,7 +111,7 @@ av_cold int CBS_FUNC(init)(CodedBitstreamContext **ctx_ptr,
     int i;
 
     type = NULL;
-    for (i = 0; i < FF_ARRAY_ELEMS(cbs_type_table); i++) {
+    for (i = 0; i < FF_ARRAY_ELEMS(cbs_type_table) && cbs_type_table[i]; i++) {
         if (cbs_type_table[i]->codec_id == codec_id) {
             type = cbs_type_table[i];
             break;
